mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -1422,14 +1442,25 @@ struct GridwiseGemmRewritePattern : public OpRewritePattern<miopen::GridwiseGemm
   LogicalResult matchAndRewrite(miopen::GridwiseGemmOp op, PatternRewriter &b) const override {
     auto loc = op.getLoc();
 
+    // Determine the type used in the filter/input/output tensors.
     auto elementType = op.output()
                            .getType()
                            .cast<MemRefType>()
                            .getElementType()
                            .template dyn_cast<Type>();
 
+    // Determine the type used on VGPR to act as accumulator.
+    // f32: f32.
+    // f16: f32 to prevent overflow from happening.
+    // i16(bf16) : i16.
```

Comment:
non-xdlops instruction doesn't support bfp16 as A and B. I think there is no need to support bfp16